src/widgets/login_window.py

```diff
@@ -68,17 +68,22 @@ def clear_entries(self, *_):
     def on_text_changed(self, entry):
         text = entry.get_text()
         can_click_next = text != ''
-        if entry.props.input_purpose == Gtk.InputPurpose.PHONE:
-            text = re.sub(r'[^+\d \-()]', '', text)
-            can_click_next = not bool(re.fullmatch(r'\D*', text))
-        elif entry.props.input_purpose == Gtk.InputPurpose.DIGITS:
-            text = re.sub(r'\D', '', text)
-            can_click_next = bool(
-                re.fullmatch('\\d{%s}' % entry.get_max_length(), text)
-            )
-        if text != entry.get_text():
-            entry.error_bell()
-            entry.set_text(text)
+        try:
+            entry_input_purpose = entry.props.input_purpose
+        except AttributeError:
+            pass
+        else:
```

Comment:
Why do you catch AttributeError? Every entry havee this `input_purpose` property

Reply:
For some reason, password entry doesn't

Reply:
Okay